cmd/snap-seccomp/versioninfo.go

```diff
@@ -0,0 +1,503 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+
+	"github.com/mvo5/libseccomp-golang"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// Generated using arch-syscall-dump test tool from libseccomp tree, git
+// revision 584ca7a5e69d87a4c2c4e4c07ce8415fa59e1351.
+var seccompSyscalls = []string{
```

Comment:
how do we know when we need to update this?

Reply:
This is a legitimate question. I would like to move this to something that is referenced in seccomp profiles so that we must introduce a new entry here in case our seccomp profiles ever want to use it.

Reply:
but does this need to be a list of things we actually use or that are supported by lib, it seems the latter (conceptually)

Reply:
We fortunately only need the list of system calls we actually use. This is why using the type system to describe the seccomp snippets and the initial templates is sufficient to ensure, at compile time, that we are only using the system calls that we are hashing here.

Reply:
Then the placement is wrong and the list should be fed from outside, this compiler does:

secSyscall, err := seccomp.GetSyscallFromName(tokens[0])

so it doesn't conceptually care about what we use or not.

Reply:
Also by definition it would die if we have a syscall that we use but the lib doesn't support it, no?  or is about things that we might optionally use?

I'm very wary of conceptual muddiness because it invites bugs and misunderstandings later.

Reply:
So to be clarify as originally discussed:

this is list is to make a fingerprint of the library , not of our usage of the library.

with that assumption my question stands

Reply:
The library error handling is incapable of differentiating unknown system calls, unsupported (on architecture) system calls from any other errors. As such we completely ignore errors in snap-seccomp today.

Reply:
> how do we know when we need to update this?

We don't. IMO the best bet is to follow the kernel and introduce new ones as they appear. `libseccomp` will eventually catch up, even if through distro backporting upstream patches.

Reply:
Honestly I'm not happy with this, could we have a spread test that takes libseccomp and builds/runs the arch-syscall-dump binary and compare this output to our list? I understand syscalls are not changing that often. But I really ffear in 1y (or 5y) noone will remember to update this list.